### ISSUE TITLE
MMLU Results

### ISSUE THREAD

**Issue by neerajgangwar** [YuanheZ/LoRA-One#3]
Hi @YuanheZ,

Thank you for the excellent work and for sharing the code to reproduce the results. I had a question regarding the MMLU results. In the paper, it is mentioned that you used direct prompting to evaluate the fine-tuned models on MMLU.

Could you kindly clarify whether direct prompting here refers to zero-shot prompting? If so, how was the model's output controlled and evaluated? Alternatively, if it refers to answer-only few-shot prompting, my understanding is that the pre-trained LLaMA2-7B typically achieves around 45% accuracy ([Reference](https://github.com/OpenNMT/OpenNMT-py/blob/master/eval_llm/MMLU/readme.md)). In that case, I am curious to know what kind of performance improvement fine-tuning on Alpaca brings. Please let me know.

Thank you!

**Comment by YuanheZ** [comment]
Hi @neerajgangwar,

Thanks for the interest. The direct prompting is specific to evaluate math reasoning via zero-shot non-CoT prompting (details refering to `gsm8k_yaml` task in https://github.com/EleutherAI/lm-evaluation-harness). For MMLU, we use answer-only 5-shot prompting to obtain the weighted average (%). In such setting, the base model achieves 45.73% which is nearly identical to vanilla LoRA's. Our method achieves 47.24% averaged over 5 runs. Here I can provide one table on each subject's score of base, LoRA, LoRA-One (seed 42).

| Subject | LoRA-One | Base | LoRA |
| :--- | :---: | :---: | :---: |
| **Specific Subjects** | | | |
| abstract_algebra | 0.350 | 0.300 | 0.330 |
| anatomy | 0.452 | 0.459 | 0.474 |
| astronomy | 0.487 | 0.401 | 0.428 |
| business_ethics | 0.510 | 0.510 | 0.500 |
| clinical_knowledge | 0.509 | 0.457 | 0.457 |
| college_biology | 0.472 | 0.465 | 0.458 |
| college_chemistry | 0.320 | 0.340 | 0.320 |
| college_computer_science | 0.420 | 0.350 | 0.360 |
| college_mathematics | 0.350 | 0.330 | 0.310 |
| college_medicine | 0.462 | 0.416 | 0.439 |
| college_physics | 0.157 | 0.235 | 0.196 |
| computer_security | 0.610 | 0.600 | 0.600 |
| conceptual_physics | 0.417 | 0.430 | 0.434 |
| econometrics | 0.325 | 0.289 | 0.360 |
| electrical_engineering | 0.497 | 0.483 | 0.407 |
| elementary_mathematics | 0.312 | 0.267 | 0.275 |
| formal_logic | 0.325 | 0.286 | 0.302 |
| global_facts | 0.320 | 0.320 | 0.300 |
| high_school_biology | 0.532 | 0.500 | 0.510 |
| high_school_chemistry | 0.345 | 0.365 | 0.340 |
| high_school_computer_science| 0.440 | 0.380 | 0.410 |
| high_school_european_history| 0.594 | 0.588 | 0.612 |
| high_school_geography | 0.566 | 0.495 | 0.535 |
| high_school_government_and_politics| 0.715 | 0.689 | 0.731 |
| high_school_macroeconomics | 0.451 | 0.459 | 0.436 |
| high_school_mathematics | 0.296 | 0.296 | 0.319 |
| high_school_microeconomics | 0.382 | 0.445 | 0.403 |
| high_school_physics | 0.344 | 0.305 | 0.285 |
| high_school_psychology | 0.650 | 0.626 | 0.639 |
| high_school_statistics | 0.306 | 0.273 | 0.310 |
| high_school_us_history | 0.588 | 0.549 | 0.608 |
| high_school_world_history | 0.667 | 0.620 | 0.662 |
| human_aging | 0.578 | 0.570 | 0.556 |
| human_sexuality | 0.542 | 0.534 | 0.527 |
| international_law | 0.636 | 0.653 | 0.628 |
| jurisprudence | 0.519 | 0.537 | 0.509 |
| logical_fallacies | 0.571 | 0.515 | 0.534 |
| machine_learning | 0.402 | 0.375 | 0.411 |
| management | 0.592 | 0.544 | 0.544 |
| marketing | 0.722 | 0.684 | 0.692 |
| medical_genetics | 0.540 | 0.530 | 0.520 |
| miscellaneous | 0.632 | 0.640 | 0.662 |
| moral_disputes | 0.529 | 0.488 | 0.494 |
| moral_scenarios | 0.254 | 0.239 | 0.239 |
| nutrition | 0.526 | 0.490 | 0.497 |
| philosophy | 0.588 | 0.592 | 0.559 |
| prehistory | 0.512 | 0.494 | 0.506 |
| professional_accounting | 0.365 | 0.358 | 0.376 |
| professional_law | 0.363 | 0.368 | 0.368 |
| professional_medicine | 0.551 | 0.511 | 0.526 |
| professional_psychology | 0.472 | 0.435 | 0.446 |
| public_relations | 0.591 | 0.555 | 0.545 |
| security_studies | 0.473 | 0.482 | 0.437 |
| sociology | 0.617 | 0.632 | 0.592 |
| us_foreign_policy | 0.640 | 0.640 | 0.650 |
| virology | 0.410 | 0.422 | 0.392 |
| world_religions | 0.684 | 0.684 | 0.673 |
| math | 0.314 | 0.285 | 0.302 |
| health | 0.511 | 0.484 | 0.486 |
| physics | 0.375 | 0.362 | 0.359 |
| business | 0.643 | 0.611 | 0.613 |
| biology | 0.513 | 0.489 | 0.493 |
| chemistry | 0.337 | 0.356 | 0.333 |
| computer science | 0.466 | 0.425 | 0.444 |
| economics | 0.410 | 0.429 | 0.414 |
| engineering | 0.497 | 0.483 | 0.407 |
| philosophy | 0.419 | 0.400 | 0.397 |
| other | 0.541 | 0.544 | 0.561 |
| history | 0.583 | 0.555 | 0.587 |
| geography | 0.566 | 0.495 | 0.535 |
| politics | 0.591 | 0.580 | 0.576 |
| psychology | 0.556 | 0.525 | 0.537 |
| culture | 0.587 | 0.593 | 0.566 |
| law | 0.391 | 0.398 | 0.394 |
| **Major Categories** | | | |
| STEM | 0.389 | 0.368 | 0.370 |
| humanities | 0.441 | 0.430 | 0.434 |
| social sciences | 0.532 | 0.519 | 0.518 |
| other (business, health, misc.)| 0.539 | 0.523 | 0.530 |
| **Overall** | | | |
| **Average accuracy** | **0.473** | 0.457 | 0.461 |

As we can observe, the improvement is a combination of significant standout performances in key subjects and broad, consistent advantages across major categories. If you have further queries, don't hesitate to post here.

**Comment by neerajgangwar** [comment]
Hi @YuanheZ,

Thanks for the clarification. Appreciate your prompt response.

Neeraj
